src/components/AddLiquidity/AddLiquidity.tsx

```diff
@@ -385,6 +383,15 @@ const AddLiquidity: React.FC<{
     setTxHash('');
   }, [onFieldAInput, txHash]);
 
+  const buttonText = useMemo(() => {
+    if (account) {
+      return error ?? 'Supply';
+    } else if (isnotMatic) {
```

Comment:
I don't think this makes sense as a function.

1. IsNotMatic internally relies on Ethereum variable so ideally that dependency should be moved in.
2. You get very little reusability out of it. 

If you want to make it an extension on Ethereum e.g.  ethereum.isNotMatic() that may make sense.